sdk/openai/azure-ai-openai-assistants/src/main/java/com/azure/ai/openai/assistants/AssistantsAsyncClient.java

```diff
@@ -3043,8 +3109,53 @@ public Mono<OpenAIFile> uploadFile(FileDetails file, FilePurpose purpose) {
             .serializeFileField("file", requestObj.getFile().getContent(), requestObj.getFile().getContentType(),
                 requestObj.getFile().getFilename())
             .serializeTextField("purpose", Objects.toString(requestObj.getPurpose()))
-            .serializeTextField("filename", requestObj.getFilename()).end().getRequestBody();
+            .serializeTextField("filename", requestObj.getFilename())
+            .end()
+            .getRequestBody();
         return uploadFileWithResponse(request, requestOptions).flatMap(FluxUtil::toMono)
             .map(protocolMethodData -> protocolMethodData.toObject(OpenAIFile.class));
     }
+
+    /**
+     * Returns information about a specific file. Does not retrieve file content.
+     * <p><strong>Response Body Schema</strong></p>
+     * 
+     * <pre>{@code
+     * byte[]
+     * }</pre>
+     *
+     * @param fileId The ID of the file to retrieve.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @return represent a byte array along with {@link Response} on successful completion of {@link Mono}.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<BinaryData>> getFileContentWithResponse(String fileId, RequestOptions requestOptions) {
```

Comment:
We should add a new feature entry in the CHANGELOG

Reply:
Oh sorry I didn't realise this was about new stuff. I thought it was just codegen related convenience methods. In that case I agree with you on maybe punting this into a separate issue.

Reply:
Issue created: https://github.com/Azure/azure-sdk-for-java/issues/39675